modules/broker-core/src/main/java/org/wso2/broker/core/Exchange.java

```diff
@@ -49,20 +59,83 @@ public static Type from(String typeString) {
 
     }
 
-    String getName();
+    private final int hashCode;
+
+    private final String name;
+
+    private final Type type;
+
+    private final BindingDao bindingDao;
+
+    private final BindingsRegistry bindingsRegistry;
+
+    protected Exchange(String name, Type type, BindingDao bindingDao) {
+        this.name = name;
+        this.type = type;
+        hashCode = name.hashCode() + type.hashCode();
+        this.bindingDao = bindingDao;
+        this.bindingsRegistry = new BindingsRegistry(this, bindingDao);
+    }
+
+    public String getName() {
+        return name;
+    }
 
-    Type getType();
+    public Type getType() {
+        return type;
+    }
 
-    void bind(Queue queue, String routingKey, FieldTable arguments) throws BrokerException;
+    public BindingDao getBindingDao() {
```

Comment:
This method is unused. Maybe we can get rid of it.

Reply:
Method removed.